openmdao/core/vec_wrapper.py

```diff
@@ -1271,11 +1272,24 @@ def _apply_unit_derivatives(self):
         """ Applies derivative of the unit conversion factor to params
         sitting in vector.
         """
+
         if self.deriv_units:
-            for name, acc in iteritems(self._dat):
-                meta = acc.meta
-                if 'unit_conv' in meta:
-                    acc.val *= meta['unit_conv'][0]
+            if self.units_cache is None:
```

Comment:
could we just call _cache_units() at the end of vecwrapper setup to avoid checking if None in each call to apply_unit_derivatives?